underscore.js

```diff
@@ -1109,12 +1109,19 @@
 
 
   // Internal recursive comparison function for `isEqual`.
-  var eq = function(a, b, aStack, bStack) {
-    // Identical objects are equal. `0 === -0`, but they aren't identical.
-    // See the [Harmony `egal` proposal](http://wiki.ecmascript.org/doku.php?id=harmony:egal).
-    if (a === b) return a !== 0 || 1 / a === 1 / b;
-    // A strict comparison is necessary because `null == undefined`.
-    if (a == null || b == null) return a === b;
+  var deepEq, eq = function(a, b, aStack, bStack) {
```

Comment:
Yes, it is. It's called from deepEq. Again, splitting eq by two functions significantly improves JIT performance, as well as inlining eq directly to _.isEqual.

This is completely unintuitive, but it's how JIT actually works. I measured every change. You could try yourself.